### ISSUE TITLE
Training on own dataset

### ISSUE THREAD

**Issue by JaninaPa** [LiQiufu/WaveSNet#2]
Hello. I was wondering whether I could train the model on my own dataset of images and masks?

**Comment by LiQiufu** [comment]
of course. You should reset some parameters, such as 'num_classes', 'in_channel', etc., before running the code on your data. But there is no information about your own dataset. So I don't know how to modify the code.

**Comment by JaninaPa** [comment]
Thanks for your fast reply.
My dataset consists of ~1200 images which contain photos of several damages on vehicles like scratches, dents etc. 
They were annotated with the VIA annotation tool (https://www.robots.ox.ac.uk/~vgg/software/via/), which saves the annotations in a json file. From this I created one folder with the original images and one folder containing the masks as binary images. So basically I want to start of having 1 class (damage) and test, whether a Wavelet Net would be helping.

The overall Use Case is to locate and classify damages into scratches, dents and cracks.

**Comment by LiQiufu** [comment]
Are you trying to classify the image pixels into foreground (damage) and background? I believe that wavelet is useful to detect the fine damage like scratches, etc., and I suggest you to apply the network "wsegnet_vgg16_bn" saved in "WaveSNet/SegNet_UNet/SegNet.py". However, before that, you should configure your own dataset file. You can follow the formation of Pascal VOC or Cityscapes saved in "WaveSNet/dataloaders/datasets".

**Comment by JaninaPa** [comment]
Thank you very much for your suggestions! Exactly that is my approach.

**Comment by JaninaPa** [comment]
I managed to train your model on my own dataset, but now i am wondering how to use the generated 4D output tensors and display it as a mask. As stated above I have 2 classes, damage and background.
The shape of the Tensor is torch.Size([1,2,256,256]). Batch Size is 1. After squeezing it into a numpy array and transposing to channels last format, I got the following values:
array[0]:
<img width="218" alt="Bildschirmfoto 2021-03-24 um 11 53 40" src="https://user-images.githubusercontent.com/58034569/112298952-97d1ee00-8c97-11eb-93e6-e9189729d8ca.png">
array[1]:
<img width="211" alt="Bildschirmfoto 2021-03-24 um 11 51 28" src="https://user-images.githubusercontent.com/58034569/112298407-49bcea80-8c97-11eb-8422-867df8d92fcf.png">
array[2]:
<img width="218" alt="Bildschirmfoto 2021-03-24 um 11 52 31" src="https://user-images.githubusercontent.com/58034569/112298547-6ce79a00-8c97-11eb-9494-1dfa1193cfa0.png">


Are these class probabilities (logits)? Do I need to pass it through  Softmax activation? Or how can I generate a mask out of this? Thanks in advance.

**Comment by LiQiufu** [comment]
er... You need more basic about PyTorch and Deep Learning. Please read the file "WaveSNet/train.py".
